### ISSUE TITLE
Scoring with ONNX does not work as expected

### ISSUE THREAD

**Issue by cosmincatalin** [dotnet/machinelearning#1549]
I have an _onnx_ model generated from an MXNet model. I try to use it for scoring a regression problem. It does not work. Here are the steps I followed:

## Build the MXNet model

```python
import pandas as pd
```

```python
df_train = pd.read_csv("train.csv")
df_test = pd.read_csv("test.csv")
```

```python
import mxnet as mx
```

```python
train_X = mx.nd.array(df_train.drop("Target", axis=1).values)
train_y = mx.nd.array(df_train.Target.values)
test_X = mx.nd.array(df_test.drop("Target", axis=1).values)
test_y = mx.nd.array(df_test.Target.values)
train_nd = list(zip(train_X, train_y))
test_nd = list(zip(test_X, test_y))
```

```python
from mxnet.gluon.data import DataLoader
```

```python
batch_size = 64
```

```python
train_data = DataLoader(train_nd, batch_size, shuffle=True)
test_data = DataLoader(test_nd, batch_size, shuffle=True)
```

```python
from mxnet.gluon.nn import HybridSequential, Dense, Dropout
from mxnet.initializer import Xavier
from mxnet.gluon.loss import L2Loss
from mxnet.gluon.trainer import Trainer
```

```python
net = HybridSequential()
with net.name_scope():
    net.add(Dense(9))
    net.add(Dropout(.25))
    net.add(Dense(16))
    net.add(Dropout(.25))
    net.add(Dense(1))
```

```python
net.hybridize()
```

```python
ctx = mx.cpu()
```

```python
net.collect_params().initialize(Xavier(magnitude=2.24), ctx=ctx)
```

```python
loss = L2Loss()
```

```python
trainer = Trainer(net.collect_params(), optimizer="adam")
```

```python
smoothing_constant = .01
epochs = 5
```

```python
def measure_performance(model, ctx, data_iter):
    mae = mx.metric.MAE()
    for _, (data, labels) in enumerate(data_iter):
        data = data.as_in_context(ctx)
        labels = labels.as_in_context(ctx)
        output = model(data)
        predictions = output
        mae.update(preds=predictions, labels=labels)
    return mae.get()[1]
```

```python
from mxnet import autograd
```

```python
for e in range(epochs):
    moving_loss = 0
    for i, (data, label) in enumerate(train_data):
        data = data.as_in_context(ctx)
        label = label.as_in_context(ctx)
        with autograd.record():
            output = net(data)
            loss_result = loss(output, label)
        loss_result.backward()
        trainer.step(batch_size)

        curr_loss = mx.nd.mean(loss_result).asscalar()
        moving_loss = (curr_loss if ((i == 0) and (e == 0))
                       else (1 - smoothing_constant) * moving_loss + smoothing_constant * curr_loss)

    test_mae = measure_performance(net, ctx, test_data)
    train_mae = measure_performance(net, ctx, train_data)
    print("Epoch %s. Loss: %s, Train_mae %s, Test_mae %s" % (e, moving_loss, train_mae, test_mae))
```
Output:

    Epoch 0. Loss: 5.848355519538663, Train_mae 1.1675882118595375, Test_mae 1.2460664133482342
    Epoch 1. Loss: 2.8617846590961733, Train_mae 0.8983533112182495, Test_mae 0.9300098409758338
    Epoch 2. Loss: 1.6959465687435336, Train_mae 0.6692642353930274, Test_mae 0.6831557900656627
    Epoch 3. Loss: 0.91648433711298, Train_mae 0.5097093659054811, Test_mae 0.5119943500885481
    Epoch 4. Loss: 0.5725724269757653, Train_mae 0.4398727014996231, Test_mae 0.4710224339667755
    

```python
import numpy as np
```

```python
net(mx.nd.array(np.expand_dims(np.ones(9), axis=0))).asnumpy().ravel().tolist()[0]
```

Output:

    7.281115531921387

```python
net.export("model", epoch=epochs - 1)
```

**7.281115531921387 is what I expect to get when I provide _ones_ as feature.**

## Export the MXNet model as ONNX

```python
from mxnet.contrib import onnx as onnx_mxnet
import numpy as np
```

```python
onnx_mxnet.export_model(sym="model-symbol.json",
                  params="model-0004.params",
                  input_shape=[(1, 9)],
                  input_type=np.float32,
                  onnx_file_path="model.onnx",
                  verbose=True)
```

Output:

    'model.onnx'

The model as visualized with Netron is as follows (the model is trivial):

![model](https://user-images.githubusercontent.com/525590/48091230-61e85800-e209-11e8-83fe-4cd66e067e2f.png)

## Testing with Tensorflow

Before loading the onnx model into ML.NET I tested it with Tensorflow

```python
import onnx
```

```python
model = onnx.load("model.onnx")
```

```python
from onnx_tf.backend import prepare
```

```python
tf_rep = prepare(model)
```

Output:

    c:\users\cosmin\dev\generate-mxnet-model\env\lib\site-packages\onnx_tf\common\handler_helper.py:74: UserWarning: Unknown op ConstantLike in domain `ai.onnx`.
      handler.ONNX_OP, handler.DOMAIN or "ai.onnx"))
    

```python
import numpy as np
```

```python
tf_rep.run(np.expand_dims(np.ones(9), axis=0)).hybridsequential0_dense2_fwd.ravel().tolist()[0]
```

Output:

    7.281115531921387

```python
tf_rep.export_graph("model.pb")
```

**As you can see running the regression on ones returns the expected result: 7.281115531921387**

## ML.NET 0.7

My programs is as follows:

`Program.cs`
```csharp
using System;
using Microsoft.ML;
using Microsoft.ML.Runtime.Data;
using Microsoft.ML.Transforms;

namespace ml.net_with_mxnet
{
    internal static class Program
    {
        private static void Main(string[] args)
        {
            var env = new MLContext();
            var reader = TextLoader.CreateReader(env,
                ctx => (
                    Feature1: ctx.LoadFloat(0),
                    Feature2: ctx.LoadFloat(1),
                    Feature3: ctx.LoadFloat(2),
                    Feature4: ctx.LoadFloat(3),
                    Feature5: ctx.LoadFloat(4),
                    Feature6: ctx.LoadFloat(5),
                    Feature7: ctx.LoadFloat(6),
                    Feature8: ctx.LoadFloat(7),
                    Feature9: ctx.LoadFloat(8),
                    Target: ctx.LoadFloat(9)),
                separator: ',',
                hasHeader: true);
            var data = reader.Read(new MultiFileSource("test.csv"));
            
            var learningPipeline = reader.MakeNewEstimator()
                .Append(row => (Target: row.Target, Features: row.Feature1.ConcatWith(
                    row.Feature2, row.Feature3, row.Feature4, row.Feature5, row.Feature6,
                    row.Feature7, row.Feature8, row.Feature9)))
                .Append(row => (Truth: row.Target, Estimate: row.Features.ApplyOnnxModel("model.onnx")));

            var model = learningPipeline.Fit(data);
            
            var predictionFunction = model.AsDynamic.MakePredictionFunction<SearchData, SearchPrediction>(env);

            var prediction = predictionFunction.Predict(new SearchData()
            {
                Feature1 = 1.0f,
                Feature2 = 1.0f,
                Feature3 = 1.0f,
                Feature4 = 1.0f,
                Feature5 = 1.0f,
                Feature6 = 1.0f,
                Feature7 = 1.0f,
                Feature8 = 1.0f,
                Feature9 = 1.0f
            });            
            Console.WriteLine(prediction.Estimate);
        }
    }
}

```

`SearchData.cs`
```csharp
using Microsoft.ML.Runtime.Api;

namespace ml.net_with_mxnet
{

    public class SearchData
    {
        [ColumnName("Target")]
        public float DummyUnused{ get; set; }
        
        public float Feature1{ get; set; }

        public float Feature2{ get; set; }

        public float Feature3{ get; set; }

        public float Feature4{ get; set; }

        public float Feature5{ get; set; }

        public float Feature6{ get; set; }

        public float Feature7{ get; set; }

        public float Feature8{ get; set; }

        public float Feature9{ get; set; }

    }

}

```

`SearchPrediction.cs`
```csharp
using Microsoft.ML.Runtime.Api;

namespace ml.net_with_mxnet
{
    public class SearchPrediction
    {
        [ColumnName("Target")]
        public float Estimate{ get; set; }
    }
}

```

When running the program, I get 0, which is not great. The training and test sets are just full of 10 columns of floats.

I also attach the `pip` requirements.txt file, so that you can see the onnx version I've used

```
absl-py==0.6.1
astor==0.7.1
backcall==0.1.0
bleach==3.0.2
certifi==2018.10.15
chardet==3.0.4
colorama==0.4.0
decorator==4.3.0
defusedxml==0.5.0
entrypoints==0.2.3
gast==0.2.0
graphviz==0.8.4
grpcio==1.16.0
h5py==2.8.0
idna==2.6
ipykernel==5.1.0
ipython==7.1.1
ipython-genutils==0.2.0
ipywidgets==7.4.2
jedi==0.13.1
Jinja2==2.10
jsonschema==2.6.0
jupyter==1.0.0
jupyter-client==5.2.3
jupyter-console==6.0.0
jupyter-core==4.4.0
Keras-Applications==1.0.6
Keras-Preprocessing==1.0.5
Markdown==3.0.1
MarkupSafe==1.0
mistune==0.8.4
mxnet==1.3.0
nbconvert==5.4.0
nbformat==4.4.0
notebook==5.7.0
numpy==1.14.6
onnx==1.3.0
onnx-tf==1.2.0
pandas==0.23.4
pandocfilters==1.4.2
parso==0.3.1
pickleshare==0.7.5
prometheus-client==0.4.2
prompt-toolkit==2.0.7
protobuf==3.6.1
Pygments==2.2.0
python-dateutil==2.7.5
pytz==2018.7
pywinpty==0.5.4
PyYAML==3.13
pyzmq==17.1.2
qtconsole==4.4.2
requests==2.18.4
Send2Trash==1.5.0
six==1.11.0
tensorboard==1.11.0
tensorflow==1.11.0
termcolor==1.1.0
terminado==0.8.1
testpath==0.4.2
tornado==5.1.1
traitlets==4.3.2
typing==3.6.6
typing-extensions==3.6.6
urllib3==1.22
wcwidth==0.1.7
webencodings==0.5.1
Werkzeug==0.14.1
widgetsnbextension==3.4.2
```

**Comment by najeeb-kazmi** [comment]
@wschin can you please give your input on this? Why would the ONNX model produce a 0 result in ML.NET when scored on a vector of ones?

**Comment by wschin** [comment]
I didn't run your code on my side but my initial guess is that Estimate is not assigned due to mis-connection between steps in your pipeline.

**Comment by cosmincatalin** [comment]
I have created a repository that outlines the above issues. It is available her: https://github.com/cosmincatalin/mxnet-onnx-mlnet

**Comment by najeeb-kazmi** [comment]
@jignparm any thoughts on this?

**Comment by codemzs** [comment]
please try with the stable release and let me know if you still see the issue.
